web/source/osk/visualKeyboard.ts

```diff
@@ -110,7 +110,16 @@ namespace com.keyman.osk {
     }
 
     set layerId(value: string) {
-      this._layerId = value;
+      const changedLayer = value != this._layerId;
+      if(this.layerGroup.layers[value]) {
+        this._layerId = value;
+      } else {
+        throw new Error(`Keyboard ${this.layoutKeyboard.id} does not have a layer with id ${value}`);
+      }
+
+      if(changedLayer) {
+        this.updateState();
+      }
```

Comment:
```suggestion
      if(!this.layerGroup.layers[value]) {
        throw new Error(`Keyboard ${this.layoutKeyboard.id} does not have a layer with id ${value}`);
      }
      if(value == this._layerId) {
        return;
      }
      this._layerId = value;
      this.updateState();
```

Suggest we try and use the [return early pattern](https://gomakethings.com/the-early-return-pattern-in-javascript/).